nixos/modules/services/misc/sonarr.nix

```diff
@@ -27,6 +36,31 @@ in
         '';
       };
 
+      settings = lib.mkOption {
+        inherit (settingsFormat) type;
+        description = "An attribute set containing Sonarr configuration settings.";
+        default = { };
+        example = lib.literalExpression ''
+          LogLevel = "info";
+          EnableSsl = "False";
+          Port = 8989;
+          SslPort = 9898;
+          UrlBase = "";
+          BindAddress = "*";
+          AuthenticationMethod = "None";
+          UpdateMechanism = "external";
+          Branch = "main";
+          InstanceName = "Sonarr";
+        '';
+      };
+
+      apiKeyFile = lib.mkOption {
+        type = lib.types.path;
```

Comment:
Nit: IMO making it nullable with a `null` default is more explicit than an empty string.